src/librustc_typeck/check/demand.rs

```diff
@@ -425,16 +425,29 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                 }
                             }
                         }
+
+                        let mut sugg = sugg_expr.as_str();
+                        if let hir::ExprKind::MethodCall(_segment, _sp, _args) = &expr.node {
+                            let clone_path = "std::clone::Clone::clone";
+                            if let Some(true) = self.tables.borrow()
+                                .type_dependent_def_id(expr.hir_id)
+                                .map(|did| self.tcx.def_path_str(did).as_str() == clone_path)
```

Comment:
I would do this as checking that this is the associated item named "clone" of the `Clone` lang item.

Reply:
Could you point me in the direction of a place where we do a similar check? I'm not sure what the appropriate API to use for that is.

Reply:
I expect it's just this:
```rust
let ai = tcx.associated_item(did);
ai.name == "clone" && ai.container == ty::TraitContainer(clone_trait_def_id)
```